components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -272,7 +281,29 @@ private void checkPasswordValidity(UserStoreException e) throws IdentityRecovery
 
             if (cause instanceof PolicyViolationException) {
                 throw IdentityException.error(IdentityRecoveryClientException.class,
-                        IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_POLICY_VIOLATION.getCode(), cause.getMessage(), e);
+                        IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_POLICY_VIOLATION.getCode(),
+                        cause.getMessage(), e);
+            }
+
+            if (StringUtils.isNotBlank(e.getMessage()) && StringUtils.indexOfAny(e.getMessage(),
+                    pwdPatternViolations) >= 0) {
+                UserStoreManager userStoreManager;
+                try {
+                    userStoreManager = IdentityRecoveryServiceDataHolder.getInstance().getRealmService().
+                            getTenantUserRealm(tenantId).getUserStoreManager();
+                } catch (UserStoreException userStoreException) {
+                    throw Utils.handleClientException(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_UNEXPECTED,
+                            null, userStoreException);
+                }
+                RealmConfiguration realmConfig = ((org.wso2.carbon.user.core.UserStoreManager)userStoreManager)
```

Comment:
It would be more readable if we could separate this part out to a method that would return the user store manager